src/overlay/OverlayManagerImpl.cpp

```diff
@@ -304,8 +304,9 @@ void
 OverlayManagerImpl::storePeerList(std::vector<std::string> const& list,
                                   bool setPreferred)
 {
-    auto typeUpgrade = setPreferred ? PeerManager::TypeUpdate::SET_PREFERRED
-                                    : PeerManager::TypeUpdate::KEEP;
+    auto typeUpgrade = setPreferred
+                           ? PeerManager::TypeUpdate::SET_PREFERRED
+                           : PeerManager::TypeUpdate::UPDATE_TO_OUTBOUND;
```

Comment:
Why do we need a new type here (and not just use `PeerManager::TypeUpdate::SET_OUTBOUND`)? @vogel @MonsieurNicolas

Reply:
SET_OUTBOUND is unconditional, UPDATE_TO_OUTBOUND keeps preferred flag - that is for case when peer that is in KNOWN_PEERS is also in PREFERRED_PEER_KEYS, so we don't know at this moment that it is preferred, but we learned about it during some previous connection after authentication phase.

Reply:
Thanks for explaining!